src/modules/musig/main_impl.h

```diff
@@ -112,14 +108,28 @@ int secp256k1_musig_pubkey_combine(const secp256k1_context* ctx, secp256k1_scrat
 
     ecmult_data.ctx = ctx;
     ecmult_data.pks = pubkeys;
+    /* No point on the curve has an X coordinate equal to 0 */
+    secp256k1_fe_set_int(&ecmult_data.second_pk_x, 0);
+    for (i = 1; i < n_pubkeys; i++) {
+        secp256k1_ge pt;
+        if (!secp256k1_xonly_pubkey_load(ctx, &pt, pubkeys[i])) {
+            return 0;
+        }
+        if (secp256k1_memcmp_var(pubkeys[0], pubkeys[i], sizeof(*pubkeys[0])) != 0) {
+            ecmult_data.second_pk_x = pt.x;
+            break;
+        }
+    }
+
     if (!secp256k1_musig_compute_ell(ctx, ecmult_data.ell, pubkeys, n_pubkeys)) {
         return 0;
     }
     if (!secp256k1_ecmult_multi_var(&ctx->error_callback, &ctx->ecmult_ctx, scratch, &pkj, NULL, secp256k1_musig_pubkey_combine_callback, (void *) &ecmult_data, n_pubkeys)) {
+        /* The current implementation of ecmult_multi_var makes this code unreachable with tests. */
         return 0;
     }
     secp256k1_ge_set_gej(&pkp, &pkj);
-    secp256k1_fe_normalize(&pkp.y);
+    secp256k1_fe_normalize_var(&pkp.y);
     pk_parity = secp256k1_extrakeys_ge_even_y(&pkp);
```

Comment:
There is something weird here.
We are returning `combined_pk` with `EvenY`, and register the `pk_parity` it was before.

However, look at `secp256k1_musig_pubkey_tweak_add`. The `output_pubkey` is not ensured to be `EvenY`.

Reply:
Yes that's what we want. The returned output_pubkey needs to be a full, non-xonly pubkey to be used in secp256k1_xonly_pubkey_tweak_add_check. In other words, we need the sign bit of `y` to be able to verify taproot tweaks. On the other hand, to verify (and sign) MuSig signatures, the 32 byte pubkey xonly pubkey is sufficient.